Sources/BrowserServicesKit/Autofill/Matchers/AutofillAccountMatcher.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  AutofillAccountMatcher.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+public struct AccountMatches {
+    public let perfectMatches: [SecureVaultModels.WebsiteAccount]
+    public let partialMatches: [String: [SecureVaultModels.WebsiteAccount]]
+
+    public init(perfectMatches: [SecureVaultModels.WebsiteAccount],
+                partialMatches: [String: [SecureVaultModels.WebsiteAccount]]) {
+        self.perfectMatches = perfectMatches
+        self.partialMatches = partialMatches
+    }
+}
+
+public protocol AutofillAccountMatcher {
+    func findMatches(accounts: [SecureVaultModels.WebsiteAccount], for url: String) -> AccountMatches
+}
+
+public struct AutofillWebsiteAccountMatcher: AutofillAccountMatcher {
+
+    private let autofillUrlMatcher: AutofillDomainNameUrlMatcher
+    private let tld: TLD
+
+    public init(autofillUrlMatcher: AutofillDomainNameUrlMatcher, tld: TLD) {
+        self.autofillUrlMatcher = autofillUrlMatcher
+        self.tld = tld
+    }
+
+    public func findMatches(accounts: [SecureVaultModels.WebsiteAccount], for url: String) -> AccountMatches {
```

Comment:
Nit: I'd be tempted to rename this to findSortedMatches to make to clear that's what it returns (or be even more specific and include what they're sorted by).
An alternative might be to rename AccountMatches if that's always sorted (although if going that way I'd be tempted to move the sorting to its constructor)